### ISSUE TITLE
version for ROS2-foxy

### ISSUE THREAD

**Issue by cf95** [basler/pylon-ros-camera#58]
Hi,

is there a similar implementation for ros2 foxy or any other ros2 version?

BR Chris

**Comment by marccastor2** [comment]
Hi, I'm interested into your question, did you find any solution?

**Comment by cf95** [comment]
Hi, I didn't find one yet, but i wasn't searching or developing one. I guess an easy solution would be to use the ros1 bridge (drawback: you would need a full ros1 installation on your device) or look into the code here, what has to be modified to get it working with ros2. Anyhow, depending on when you would need it, I have to look into this for my master thesis, but it may take a month or two until I get to the point, where I really need it. I can then report my solution here

**Comment by pablo-quilez** [comment]
Hello,

I think makes a lot of sense to port the driver to ROS2. However, the necessary effort to do it is relative high. Therefore it is not a task which can start immediately. I keep open the issue for the future.

Cheers,
Pablo

**Comment by bigrobinson** [comment]
I second this emotion. ROS2 drivers would be great.

**Comment by MCP-TTC** [comment]
I agree, a ROS2 driver is a key component for ROS2 in industrial applications

**Comment by jmarsik** [comment]
Is there any progress in this matter? We would also benefit from having a ROS2 native driver for Basler cameras.

**Comment by fhwedel-hoe** [comment]
I could use some input regarding my driver, especially when it comes to performance. It does not support any kind of dynamic reconfiguration so it might be unsuitable for your needs.

**Comment by jmarsik** [comment]
@fhwedel-hoe I think it would be ok. But our camera is connected through GigE Vision :(

**Comment by jmarsik** [comment]
@fhwedel-hoe would it be complicated to add basic support for GigE Vision cameras to your driver?

**Comment by fhwedel-hoe** [comment]
@jmarsik Code-wise, it is trivial. I replaced "Usb" with "GigE" and https://github.com/fhwedel-hoe/pylon_usb-instant-camera/tree/ros2_gige compiles. Since I do not have a Basler GigE camera at my hands, I cannot run tests so I have no idea whether this actually works. Feel free to give it a try (the package itself is still called pylon_usb_instant_camera with the PylonUSBCameraNode – I did not make the replacement in the names).
